pkg/kubernetes/client.go

```diff
@@ -15,3 +24,47 @@ func GetDiscoveryClient(mgr manager.Manager) (*discovery.DiscoveryClient, error)
 	}
 	return dc, nil
 }
+
+// DeleteAndWait deletes the given runtime object and waits until it is fully deleted
+func DeleteAndWait(c client.Client, obj runtime.Object, options ...client.DeleteOption) error {
```

Comment:
what do you think about a Delete(c, obj, `wait bool`, options) ?

Reply:
I'd rather add another func 

```
Delete(c client.Client, obj runtime.Object, options ...client.DeleteOption) error
```

which does not wait. I think having the "AndWait" in the name makes the intention more clear than a bool param.